src/MacroExpander.js

```diff
@@ -282,6 +310,26 @@ export default class MacroExpander implements MacroContextInterface {
         return output;
     }
 
+    /**
+     * Fully expand the given token stream and return the resulting list of tokens
+     */
+    expandTokens(tokens: Token[]): Token[] {
+        const output = [];
+        const oldStackLength = this.stack.length;
+        tokens = tokens.slice(); // make a shallow copy
+        tokens.reverse();
+        this.pushTokens(tokens);
+        while (this.stack.length > oldStackLength) {
+            // expand the token following \noexpand to itself
+            const expanded = this.expandOnce(true);
```

Comment:
TODO: Primitive commands such as `\def` and `MathSymbol` defined using macros should not be expanded here.